### PR TITLE
planner: add more test cases about tiflash and dynamic mode

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: related to #24150 

Problem Summary: planner: add more test cases about tiflash and dynamic mode

### What is changed and how it works?

planner: add more test cases about tiflash and dynamic mode

### Check List 

Tests 

- Unit test

### Release note 

- planner: add more test cases about tiflash and dynamic mode

### CODE REVIEW

File: executor/tiflash_test.go [hunk 1]
```diff
@@ -248,6 +250,140 @@ func (s *tiflashTestSuite) TestInjectExtraProj(c *C) {
...
+	vals := make([]string, 0, 100)
+	for i := 0; i < 100; i++ {
+		vals = append(vals, fmt.Sprintf("(%v, %v)", rand.Intn(400), rand.Intn(400)))
+	}
+	for _, tbl := range []string{`thash`, `trange`, `tlist`, `tnormal`} {
+		tk.MustExec(fmt.Sprintf("insert into %v values %v", tbl, strings.Join(vals, ", ")))
+		tk.MustExec(fmt.Sprintf("analyze table %v", tbl))
+	}
+
+	tk.MustExec("SET tidb_opt_broadcast_join=0")
```
Comment (hanfei1991) [inline executor/tiflash_test.go:294]: shall we check failpoint or use tidb_allow_mpp=2 to make sure we use mpp?

File: executor/tiflash_test.go [hunk 2]
```diff
@@ -248,6 +250,140 @@ func (s *tiflashTestSuite) TestInjectExtraProj(c *C) {
...
+
+	vals := make([]string, 0, 500)
+	for i := 0; i < 500; i++ {
+		vals = append(vals, fmt.Sprintf("(%v, %v)", rand.Intn(400), rand.Intn(400)))
+	}
+	for _, tbl := range []string{`thash`, `trange`, `tlist`, `tnormal`} {
+		tk.MustExec(fmt.Sprintf("insert into %v values %v", tbl, strings.Join(vals, ", ")))
+	}
+
+	tk.MustExec("set @@session.tidb_isolation_read_engines='tiflash'")
```
Comment (hanfei1991) [inline executor/tiflash_test.go:365]: ditto.

File: executor/tiflash_test.go [hunk 3]
```diff
@@ -248,6 +250,140 @@ func (s *tiflashTestSuite) TestInjectExtraProj(c *C) {
...
+		idx := i % 4
+		if listPartitions[idx] != "" {
+			listPartitions[idx] += ", "
+		}
+		listPartitions[idx] = listPartitions[idx] + fmt.Sprintf("%v", i)
+	}
+	tk.MustExec(`create table tlist (a int, b int) partition by list(a) (
+		partition p0 values in (` + listPartitions[0] + `), partition p1 values in (` + listPartitions[1] + `),
+		partition p2 values in (` + listPartitions[2] + `), partition p3 values in (` + listPartitions[3] + `))`)
+	tk.MustExec(`create table tnormal (a int, b int) partition by hash(a) partitions 4`)
```
Comment (rebelice) [inline executor/tiflash_test.go:276]:
```suggestion
	tk.MustExec(`create table tnormal (a int, b int)`)
```

**Review by rebelice — APPROVED** [review]

**Review by hanfei1991 — APPROVED** [review]

**Pull request merged**